earth_enterprise/rpms/build.gradle

```diff
@@ -89,34 +89,6 @@ def rpmPlatformString = org.opengee.os.Platform.rpmPlatformString
 // Create a file handle for gee_long_version_txt
 def openGeeVersionFile = new File( gee_long_version_txt )
 
-// Call scons to create the file 'gee_long_version.txt'
-// The actual parameter used by scons is based on the parameter
-// passing into Gradlew
-if ( !openGeeVersionFile.exists() ) {
-    GeeCommandLine.expand(
-      ["/usr/bin/scons", scons_option, gee_long_version_txt ],
-      "Runing `scons` to produce <${gee_long_version_txt}> failed!",
-      null, // Use default environment variables.
-      new File( src_dir )
-    )
-}
-
-def openGeeVersion =
-    openGeeVersionFile.readLines().
-    collect {
-        // Strip comments:
-        def commentIndex = it.indexOf('#')
-
-        commentIndex < 0 ? it : it.substring(0, commentIndex)
-    }.
-    collect { it.trim() }. // Trim whitespace.
-    findAll { !it.isEmpty() }. // Skip empty lines.
-    findResult {
-        it // The first string that remains is the version string.
-    }.
-    replaceAll(/-.*release/, '').
-    replaceAll('-', '.')
```

Comment:
You're not running the code to stage an install, or build Open GEE.  It's just the code to output the version file.

Reply:
I think it would be normal for a `clean` task to rely on calculating the output files from all other tasks, which would require having the version.  It's true that the `clean` operations that apply to most plugins operations are not that fine-grained, and would just delete the `build` directory without having to know the output files from each task.  Obviously, that wouldn't work in general, since there isn't anything restricting you from generating files in any arbitrary location from a task.  So, in this case, running a `clean` without version would happen to work, but in general, that wouldn't be the usual way it operates.

Task creation is also dynamic in Gradle (which is why the configuration stage needs to complete to know what tasks exist).  For the sake of argument, you could, for example, have some tasks that only existed for GEE version 5.x.x, and not for GEE version 6.x.x.  Again, we're not doing anything like that, so we can get away with using a dummy value for the GEE version, but, in general, I can make sense that you need such information just to run `gradle tasks`.

Yes, I think staging would normally depend on generating version information, so if would be normal for staging to call the tasks that output that information.  I also don't see anything wrong with getting that information before staging.  To build an RPM you need both version information and all the files that would be installed.  But when you need version information, you should perform the operations to get version information, not some other operation that also happens to generate version information.

Reply:
The generic clean task in gradle is really just a place holder tasks for others to extend.  If you create a task that creates output you *should* extend the clean task to do proper clean up.  Many don't and many just extend clean to delete the build folder because of that.  

The reality is currently there is nothing that uses the version information until execution phase.  But yes, that could in theory change later.

Reply:
Oh, yes, I have a clean task:

![under the rug](https://farm8.staticflickr.com/7279/7695348682_1a094de51d_q.jpg)